components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java

```diff
@@ -99,4 +99,18 @@ public void deleteCache(Context context, String subscriptionKey) {
             Log.e(TAG, "Failed to open cache directory when deleting cache.");
         }
     }
+
+    private void sleepIfTesting(Context context, boolean shouldSleep) {
```

Comment:
now the production code will have test code. Maybe you can pass a nullable lambda here and runs it when it's not null.